doc/source/serve/deployment-graph.md

```diff
@@ -0,0 +1,566 @@
+- In production, the deployments making up the graph can be reconfigured and scaled dynamically. This should enable DevOps/MLOps teams to operate deployment graphs without modifying the underlying code.
+
+
+__[Full Ray Enhancement Proposal, REP-001: Serve Pipeline](https://github.com/ray-project/enhancements/blob/main/reps/2022-03-08-serve_pipeline.md)__
+
++++
+
+## Concepts
+
+- **Deployment**: Scalable, upgradeable group of actors managed by Ray Serve. __[See docs for detail](https://docs.ray.io/en/master/serve/core-apis.html#core-api-deployments)__
+
+- **DeploymentNode**: Smallest unit in a graph, created by calling `.bind()` on a serve decorated class or function, backed by a Deployment.
+
+- **InputNode**: A special node that represents the input passed to a graph at runtime.
+
+- **Deployment Graph**: Collection of deployment nodes bound together to define an inference graph. The graph can be deployed behind an HTTP endpoint and reconfigured/scaled dynamically.
+
++++
+
+## Full End to End Example Walkthrough
+
+Let's put the concepts together and incrementally build a runnable DAG example highlighting the following features:
+
+```{tip}
+At the end of this document we have the full and end to end executable implementation.
+```
+
+- Building a graph:
+    - A deployment node is created from `@serve.deployment` decorated function or class. 
+    - You can construct different deployment nodes from same class or function.
+    - Deployment nodes can be used as input args in .bind() of other nodes in the DAG. 
+    - Multiple nodes in the deployment graph naturally forms a DAG structure.
+- Accessing input:
+    - Same input or output can be used in multiple nodes in the DAG.
+    - Deployment nodes can access partial user input.
+- Dynamic control flow:
+    - A deployment node can call into other nodes in the deployment graph.
+    - You can use the dynamic calling to perform control flow operation that's hard to expressive in traditional DAG.
+- Running a DAG:
+    - Nodes in the graph, such as functions or class methods, can be either sync or async.
+
++++
+
+![deployment graph](https://github.com/ray-project/images/blob/master/docs/serve/deployment_graph.png?raw=true)
+
++++
+
+### Step 1: User InputNode and preprocessor 
+
+Let's start with the first layer of DAG: Building user input to two preprocessor functions, where each function receives parts of the input data. For simplicity, we use the same existing `@serve.deployment` decorator on an async function body.
+
++++
+
+```python
+from ray import serve
+# We will later move Ray DAG related components 
+# out of experimental in later stable release
+from ray.experimental.dag.input_node import InputNode
+
+@serve.deployment
+async def preprocessor(input_data: str):
+    """Simple feature processing that converts str to int"""
+    time.sleep(0.1) # Manual delay for blocking computation
```

Comment:
still looks like this is `time.sleep` ?

Reply:
ah i forgot to update this one. pushing new commit now